### ISSUE TITLE
react-native link is not adding libraries to project - Mach-O linker error

### ISSUE THREAD

**Issue by phil-andrews** [OneSignal/react-native-onesignal#401]
I had to go in and manually add and link libraries after trying to `react-native link` multiple times. It's worth nothing that this resulted in the infamous `Mach-O linker` error being thrown.

**Comment by Nightsd01** [comment]
Can you add some more details please @phil-andrews ? 

For example, can you copy and paste the compiler error you are getting in Xcode? How did you manually add the library? What errors were you seeing when you ran `react-native link`...? It is very difficult to fix/help with an issue with such little detail. Thanks.

**Comment by Nightsd01** [comment]
@phil-andrews Closing this issue due to inactivity. Please feel free to add more details otherwise I can't do much to help you. I'd be happy to reopen the issue and investigate if you do.

**Comment by phil-andrews** [comment]
I'm sorry, been grinding on a project. I can't remember what the error was aside from the basic overarching Mach-O category. However, I can point out that the only part of the`react-native link` process that did not work was in adding `libRCTOneSignal.a`. I had to go in and add that manually. Aside from that everything else was as it should be, header search paths were correct, files were added, etc.
